File: C#/unique-number-of-occurrences.cs
public class Solution {
    public bool UniqueOccurrences(int[] arr) {
        Dictionary<int,int> count = new();
        foreach (int n in arr) {
            if (!count.ContainsKey(n)) {
                count[n] = 0;
            }
            count[n]++;
        }
        HashSet<int> seen = new();
        foreach (var (_, n) in count) {
            if (seen.Contains(n)) {
                return false;
            }
            seen.Add(n);
        }
        return true;
    }
}

File: Go/unique-number-of-occurrences.go
func uniqueOccurrences(arr []int) bool {
    count := map[int]int{}
    for _, n := range arr {
        count[n]++
    }
    seen := map[int]bool{}
    for _, n := range count {
        if seen[n] {
            return false
        }
        seen[n] = true
    }
    return true
}